readthedocs/index.rst

```diff
@@ -14,29 +14,84 @@ Word2Vec based similarity measure of mass spectrometry data.
+To get a model that can give a meaningful representation of a set of
+given spectra it is desirable to train the model on a large and representative
+dataset.
 
 .. code-block:: python
 
+    import os
     import gensim
+    from matchms import calculate_scores_parallel
+    from matchms.filtering import add_losses
+    from matchms.filtering import add_parent_mass
+    from matchms.filtering import default_filters
+    from matchms.filtering import normalize_intensities
+    from matchms.filtering import reduce_to_number_of_peaks
+    from matchms.filtering import require_minimum_number_of_peaks
+    from matchms.filtering import select_by_mz
+    from matchms.importing import load_from_mgf
     from spec2vec import Spec2VecParallel
     from spec2vec import SpectrumDocument
 
-    # reference_spectrums & query_spectrums loaded from files using https://matchms.readthedocs.io/en/latest/api/matchms.importing.load_from_mgf.html
-    references = [SpectrumDocument(s, n_decimals=2) for s in reference_spectrums]
-    queries = [SpectrumDocument(s, n_decimals=2) for s in query_spectrums]
+    def apply_my_filters(s):
+        """This is how a user would typically design his own pre- and post-
+        processing pipeline."""
+        s = default_filters(s)
+        s = add_parent_mass(s)
+        s = normalize_intensities(s)
+        s = reduce_to_number_of_peaks(s, n_required=10, ratio_desired=0.5)
+        s = select_by_mz(s, mz_from=0, mz_to=1000)
+        s = add_losses(s, loss_mz_from=10.0, loss_mz_to=200.0)
+        s = require_minimum_number_of_peaks(s, n_required=10)
+        return s
+
+    # Load data from MGF file and apply filters
+    spectrums = [spectrum_processing(s) for s in load_from_mgf("reference_spectrums.mgf")]
+
+    # Omit spectrums that didn't qualify for analysis
+    spectrums = [s for s in spectrums if s is not None]
+
+    documents = [SpectrumDocument(s) for s in spectrums]
+
+    model_file = "references.model"
+    model = train_new_word2vec_model(documents, model_file, iterations=[10, 20, 30],
+                                     workers=2, progress_logger=True)
+
+Derive spec2vec similarity scores
+*********************************
+Once a word2vec model has been trained, spec2vec allows to calculate the similarities
+between mass spectrums based on this model. In cases where the word2vec model was
+trained on data different than the data it is applied for, a number of peaks ("words")
+might be unknown to the model (if they weren't part of the training dataset). To
+account for those cases it is important to specify the ``allowed_missing_percentage``,
+as in the example below.
+
+.. code-block:: python
+
+    # query_spectrums loaded from files using https://matchms.readthedocs.io/en/latest/api/matchms.importing.load_from_mgf.html
```

Comment:
This code block is missing imports. Could you move the unused from code block above to here.

Reply:
That's indeed clearer, thanks. 
I changed that and also fixed a error I had in the second code blog.